### PR TITLE
Bump controller runtime to v0.15.0

### PR DESCRIPTION
**Pull request by kkkkun**
xref  https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/pull/335

Upgrade controller-runtime to v0.15.0, which is block https://github.com/kubernetes-sigs/kubebuilder/pull/3421

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes # https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/issues/333

**Special notes for your reviewer**:

**Additional documentation**:

### CODE REVIEW

**Comment by atoato88** [comment]
/ok-to-test
I added a label to test. By this, we can check the result by prow on this PR.

File: examples/guestbook-operator/controllers/guestbook_controller.go [hunk 1]
```diff
@@ -78,14 +78,13 @@ func (r *GuestbookReconciler) SetupWithManager(mgr ctrl.Manager) error {
...
-	err = c.Watch(&source.Kind{Type: &api.Guestbook{}}, &handler.EnqueueRequestForObject{})
+	err = c.Watch(source.Kind(mgr.GetCache(), &api.Guestbook{}), &handler.EnqueueRequestForObject{})
 	if err != nil {
 		return err
 	}
 
 	// Watch for changes to deployed objects
-	_, err = declarative.WatchChildren(declarative.WatchChildrenOptions{Manager: mgr, Controller: c, Reconciler: r, LabelMaker: r.watchLabels})
-	if err != nil {
+	err = declarative.WatchChildren(declarative.WatchChildrenOptions{Manager: mgr, Controller: c, Reconciler: r, LabelMaker: r.watchLabels})	if err != nil {
```
Comment (atoato88) [inline examples/guestbook-operator/controllers/guestbook_controller.go:87]:
```suggestion
	err = declarative.WatchChildren(declarative.WatchChildrenOptions{Manager: mgr, Controller: c, Reconciler: r, LabelMaker: r.watchLabels})
	if err != nil {
```
There is a syntax error.

**Review by atoato88 — COMMENTED** [review]
I left a comment.

**Comment by kkkkun** [comment]
> I left a comment.

The tests passed. Thanks.

**Comment by atoato88** [comment]
I have informed @justinsb on [#335](https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/pull/335#issuecomment-1544968824) , but I would like to confirm with him regarding the validity of this revision.
I apologize for repeating my statement.

**Comment by kkkkun** [comment]
We need this PR for kubebuider. Could you please review this @stephanhorsthemke @camilamacedo86

**Comment by atoato88** [comment]
/lgtm

There have been no changes for about 1 week.
We need to move forward to release kubebuilder.

/assign @justinsb 
Gentle ping.
Could you review this PR?

**Comment by atoato88** [comment]
/approve
I acknowledge that we have not made progress, but we need to move forward.

@justinsb 
I apologize for approving this PR without your comments. Please feel free to comment if you have any.

**Comment by kkkkun** [comment]
We need a tag.

**Comment by atoato88** [comment]
@justinsb 
Could you add a tag against the current content?

**Comment by kkkkun** [comment]
> @justinsb Could you add a tag against the current content?
Actually it's not need tag. It's used commit id. 
https://github.com/kubernetes-sigs/kubebuilder/pull/3421/commits/d5ff1ae4153a89f162458d0a6552fbfe37b679a1

**Pull request merged**